### DISCUSSION TITLE
Controlling auto-generated library files

### DISCUSSION THREAD

**Discussion by metanomial** [beakerbrowser/beaker#1549] [General] [answered]
Is there a way to turn off auto-generation of `.datignore` and `dat.json` files in new libraries? I don't typically find `.datignore` useful, and I'd prefer to prepare and add a `dat.json` in one go instead of having two unnecessary revisions.

**Comment by pfrazee** [comment, marked as answer]
In the upcoming 1.0 release, only the `index.json` is created automatically. Are you looking for a way to not generate that file as well?

**Reply by metanomial** [reply]
Good to hear! If I could set a title and description for `index.json` before it gets written to the archive, that satisfies my typical use case. Even though it's a small thing, it just bugs me that the first three revisions (as of v0.8) are useless files.

But yes, there are use cases for archives without any indexing data. Is there a way to disable that in v1.0?

**Reply by pfrazee** [reply]
We could look into not creating the `index.json` if no title or description is set
